libp2p/src/main/kotlin/io/libp2p/security/tls/TLSSecureChannel.kt

```diff
@@ -132,8 +139,15 @@ fun buildTlsHandler(
                 cause = cause.cause
             handshakeComplete.completeExceptionally(cause)
         } else {
-            val negotiatedProtocols = sslContext.applicationProtocolNegotiator().protocols()
-            val selectedMuxer = muxers.findBestMatch(negotiatedProtocols)
+            val nextProtocol = handler.applicationProtocol()
+            val selectedMuxer = muxers.stream()
+                .flatMap { mux ->
+                    mux.protocolDescriptor.announceProtocols.stream()
+                        .filter({ it.equals(nextProtocol) })
+                        .map { NegotiatedStreamMuxer(mux, it) }
+                }
+                .findFirst()
+                .orElse(null)
```

Comment:
```suggestion
            val selectedMuxer = muxers
                .filter { mux ->
                    mux.protocolDescriptor.protocolMatcher.matches(nextProtocol)
                }
                .map { mux ->
                    NegotiatedStreamMuxer(mux, nextProtocol) 
                }
                .firstOrNull()
```
It's better use `protocolDescriptor.protocolMatcher` instead of strict `ProtocolId` equals as it may potentially may differ (not yet the case if I remember correctly)

+ also use Kotlin collections instead of Java streams